osm2streets/src/road.rs

```diff
@@ -443,6 +455,45 @@ impl Road {
     pub fn common_endpoint(&self, other: &Road) -> CommonEndpoint {
         CommonEndpoint::new((self.src_i, self.dst_i), (other.src_i, other.dst_i))
     }
+
+    /// This trims a polyline on both ends. Positive trim distances mean to shorten the polyline,
+    /// and negative mean to extend the polyline in a straight line by some amount. If this returns
+    /// `None`, then the two trim distances are incompatible -- the entire polyline disappears.
+    ///
+    /// TODO Maybe move this to PolyLine directly. This is a utility method currently for A/B
+    /// Street road editing to also use.
+    pub fn trim_polyline_both_ends(
+        mut pl: PolyLine,
+        trim_start: Distance,
+        trim_end: Distance,
+    ) -> Option<PolyLine> {
+        // The two ends trimmed past each other
+        if trim_start + trim_end > pl.length() {
+            return None;
+        }
+
+        // Note we use maybe_exact_slice and bail out upon failure of the resulting line being too
```

Comment:
Aka, this is the epsilon check part of the above. `pl.length() - trim_start - trim_end` is nonzero but too small for geom layer